### ISSUE TITLE
Incorrect window border scaling when moving between monitors.

### ISSUE THREAD

**Issue by KGB-8375** [MultiMC/MultiMC5#3666]
OS:
Windows 10, x64

Steps to reproduce:
-open multimc on one monitor
-move the window to a secondary monitor of a different resolution

Description of Issue:
Multimc's UI itself scales fine, however, the window border (mostly the title bar) becomes the wrong size. I use a 4k monitor as the primary one, so when I move it to my secondary 1080p monitor it's 4x larger than it should be.

**Comment by peterix** [comment]
MultiMC does nothing with title bars on any OS. They are the responsibility of whatever window manager you are using.

So, Windows bug?

**Comment by AverageComet250** [comment]
Yes and no. I've seen this too, most recently yesterday and it's something I've _always_ seen. It's also never happened with another app, so it's either a multimc or Qt bug

**Comment by KGB-8375** [comment]
This is a fairly common bug for windows programs especially Qt ones, so I'm not sue. I believe Okular by KDE fixed this somewhat recently in their program.

**Comment by phit** [comment]
this looks like the commit in question https://github.com/KDE/okular/commit/3311bcb949a51f1004491e5857117e30a08297ce
could be worth a shot?

**Comment by KGB-8375** [comment]
Judging by this stackoverflow post
https://stackoverflow.com/questions/35816944/improper-scaling-of-qt-desktop-application-in-windows-10

that might actually fix the problem

**Comment by KGB-8375** [comment]
this also has a similar approach
https://vicrucann.github.io/tutorials/osg-qt-high-dpi/

> 
> 
> Judging by this stackoverflow post
> https://stackoverflow.com/questions/35816944/improper-scaling-of-qt-desktop-application-in-windows-10
> 
> that might actually fix the problem

**Comment by peterix** [comment]
> This enum was introduced or modified in Qt 5.14.

We use 5.6

**Comment by peterix** [comment]
Also, most of this does not apply.

**Comment by KGB-8375** [comment]
> 
> 
> > This enum was introduced or modified in Qt 5.14.
> 
> We use 5.6

Both of these tutorials are explicitly for 5.6, so I don't know why you're saying this. I'm going to attempt to build the source myself and test the Microsoft SetProcessDPIAware(); function to see if this would fix my issue.

**Comment by KGB-8375** [comment]
OK I fixed this issue on my PC (I'm not sure how to implement this in code since I'm not great with C++ or anything, it seems like mingw hasn't implemented any compatibility for the SetProcessDPIAware() function).

Steps to fix:
find multimc.exe
right click, select Properties
go to Compatibility tab
click Change high DPI settings
check Override high DPI scaling behaviour
select "System" or "System (enhanced)" on the dropdown box

This does make the application's text render in 1080p on the 4k monitor however, causing it to look slightly blurry.

**Comment by KGB-8375** [comment]
Actually, I've fixed this problem properly and created a [pull request](https://github.com/MultiMC/MultiMC5/pull/3684).

**Comment by peterix** [comment]
OK, that definitely was not a fix.

**Comment by KGB-8375** [comment]
Dang I don't know why it wouldn't work on his version. Do you know where I can download the development version to test myself? It should be almost identical since I have a very very similar setup to him (except with 2 monitors and 175% scaling on the 4k one). I almost wonder if it's not finding the user32.dll on his computer or something.

**Comment by KGB-8375** [comment]
For reference here is what it looks like on the current dev version (now that it's been reverted

![incorrect_scaling](https://user-images.githubusercontent.com/60853993/112762756-704c8f80-8fcf-11eb-9c76-5868198d944c.PNG)
)

**Comment by peterix** [comment]
Is that a normal version of Windows 10, or some development preview?

**Comment by KGB-8375** [comment]
The normal version of windows 10. And like I said earlier, the function is meant to work on every windows since vista (and is part of a core system library)

**Comment by KGB-8375** [comment]
Once the windows DPI functionality takes over it works perfectly fine as far as I can tell on both monitors
![patched_version](https://user-images.githubusercontent.com/60853993/112763261-b99dde80-8fd1-11eb-800b-459e0ed18af1.PNG)

**Comment by KGB-8375** [comment]
My theory is it somehow can't activate the function so it turns of Qt's dpi scaling without turning on window's

**Comment by peterix** [comment]
My theory is that there is absolutely no point looking at this without updating Qt to get rid of all the bugs.

**Comment by peterix** [comment]
Example: Qt 5.6 might be scale aware, but it does this, and only to fonts...
![image](https://user-images.githubusercontent.com/203326/112763529-18308000-9005-11eb-830b-89d5445739e5.png)

**Comment by peterix** [comment]
We would have to drop XP and Vista support

**Comment by peterix** [comment]
Try this one.

Does this work for you?
[MultiMC-Qt-5.15.2.zip](https://github.com/MultiMC/MultiMC5/files/6218242/MultiMC-Qt-5.15.2.zip)

**Comment by KGB-8375** [comment]
Yes, that one seems to work perfectly for me...

**Comment by AverageComet250** [comment]
@peterix does that mean that multimc is moving Qt versions to fix this?

**Comment by ThexXTURBOXx** [comment]
> 
> 
> Try this one.
> 
> Does this work for you?
> [MultiMC-Qt-5.15.2.zip](https://github.com/MultiMC/MultiMC5/files/6218242/MultiMC-Qt-5.15.2.zip)

Yes, that one works also for me (I was the one guy from #3697)

**Comment by ThexXTURBOXx** [comment]
Okay, Update: It works on my main screen properly:

![grafik](https://user-images.githubusercontent.com/12763829/112847190-1ae3b180-90a7-11eb-8c60-5cdd729651a1.png)

On my secondary screens, some icons screw up:

![grafik](https://user-images.githubusercontent.com/12763829/112847276-35b62600-90a7-11eb-9f43-08f87757ae3e.png)

**Comment by KGB-8375** [comment]
It seems that only the default icon set is scaling properly on the Qt 1.15.2 version

**Comment by peterix** [comment]
Amazing. That is definitely a Qt bug :D

**Comment by AverageComet250** [comment]
This is why I prefer win32 over qt

Thanks,
Shravan Mandava

________________________________

**Comment by peterix** [comment]
Win32 is not viable in 2021 and Linux is the main development platform for MultiMC

**Comment by AverageComet250** [comment]
It's a joke I'm not telling you to move MMC to win32 as that's just dumb.

**Comment by peterix** [comment]
In any case, SVG icons have a problem in toolbars... Which is weird to see.

**Comment by phit** [comment]
some more issues the Qt update, it messed up the system theme

new ... old
![image](https://user-images.githubusercontent.com/2097483/113518745-74a51980-9588-11eb-8ed8-7e25c5f911a3.png)

**Comment by AverageComet250** [comment]
Can we have a progress update on this, cause this is still the only major issue I have with MultiMC?

**Comment by peterix** [comment]
There is no progress.

**Comment by Alex0322** [comment]
Compatibility mode "Windows 7" for MultiMC.exe helps me with 150% dpi fonts
